### PR TITLE
Handle NaN from CloseLocationValueIndicator as zero

### LINKED ISSUE #375: CloseLocationValueIndicator returns NaN if max price = min price
CloseLocationValueIndicator returns NaN if max price = min price, since it's dividing 0
max price could equal to min price for 1 min chart during quiet hours 
due to the cached nature of EMA, it means all subsequent values will all become NaN for indicators like ChaikinOscillatorIndicator

### PR DESCRIPTION
**Pull request by team172011**
Fixes #375.

Changes proposed in this pull request:
- CloseLocationValueIndicator returns zero instead of NaN if high==low

- [x] added an entry to the unreleased section of `CHANGES.md`

### CODE REVIEW

**Comment by team172011** [comment]
Maybe we should just fix the `CloseValueIndicators` returning zero instead of NaN?

**Comment by nimo23** [comment]
I guess, returning zero for such cases is the right way:

- maxPrice>minPrice -> return > 0
- maxPrice<minPrice -> return < 0
- maxPrice=minPrice -> return 0

**Comment by nimo23** [comment]
> Maybe we should just fix the CloseValueIndicators returning zero instead of NaN?

There is no `CloseValueIndicator`. You mean `CloseLocationValueIndicator`?

**Comment by nimo23** [comment]
>returning zero instead of NaN

Yes.

Normally, CLV returns 0, only if the close is exactly halfway between high and low.
However, it is not wrong to return 0 if open=high=low=close.

File: ta4j-core/src/main/java/org/ta4j/core/indicators/helpers/CloseLocationValueIndicator.java [hunk 1]
```diff
@@ -40,6 +40,11 @@ public CloseLocationValueIndicator(TimeSeries series) {
         super(series);
     }
 
+    /**
+     *
+     * @param index the bar index
+     * @return the CLC value or {@link org.ta4j.core.num.NaN} if high == low
```
Comment (milczarekIT) [inline ta4j-core/src/main/java/org/ta4j/core/indicators/helpers/CloseLocationValueIndicator.java]: should be 'the CL**V** value'

File: ta4j-core/src/main/java/org/ta4j/core/indicators/helpers/CloseLocationValueIndicator.java [hunk 2]
```diff
@@ -40,6 +40,11 @@ public CloseLocationValueIndicator(TimeSeries series) {
...
+     */
     @Override
     protected Num calculate(int index) {
```
Comment (milczarekIT) [inline ta4j-core/src/main/java/org/ta4j/core/indicators/helpers/CloseLocationValueIndicator.java:46]: CLV oscillates around 0 (zero) in the range from -1 to +1.
NaN is not a correct value for this indicator. For this case when high == low, 0 should be returned.

It is exactly the same case when we have balanced bullish and bearish pressure. Neither Bulls nor Bears were able to push price neither up nor down from the middle level.

Please add missing test cases in CloseLocationValueIndicatorTest:
1) high == low
2) close - low == high - close (e.g. low: 100, close: 120, high: 140)

File: ta4j-core/src/test/java/org/ta4j/core/indicators/volume/ChaikinMoneyFlowIndicatorTest.java [hunk 3]
```diff
@@ -33,12 +33,12 @@
 
 public class ChaikinMoneyFlowIndicatorTest {
 
+    ZonedDateTime now = ZonedDateTime.now();
```
Comment (milczarekIT) [inline ta4j-core/src/test/java/org/ta4j/core/indicators/volume/ChaikinMoneyFlowIndicatorTest.java]: Access can be private

**Comment by milczarekIT** [comment]
Hi
@nimo23  is right. 0 should be returned.

@team172011 - implementation is not correct. Please fix it. 
Also, I requested for missing tests:
high == low
close - low == high - close (e.g. low: 100, close: 120, high: 140)

In both cases, 0 should be returned.

**Pull request merged**